table/tables/tables.go

```diff
@@ -51,17 +51,19 @@ import (
 type TableCommon struct {
 	tableID int64
 	// physicalTableID is a unique int64 to identify a physical table.
-	physicalTableID int64
-	Columns         []*table.Column
-	PublicColumns   []*table.Column
-	VisibleColumns  []*table.Column
-	HiddenColumns   []*table.Column
-	WritableColumns []*table.Column
-	writableIndices []table.Index
-	indices         []table.Index
-	meta            *model.TableInfo
-	allocs          autoid.Allocators
-	sequence        *sequenceCommon
+	physicalTableID   int64
+	Columns           []*table.Column
+	PublicColumns     []*table.Column
+	VisibleColumns    []*table.Column
+	HiddenColumns     []*table.Column
+	WritableColumns   []*table.Column
+	writableIndices   []table.Index
+	indices           []table.Index
+	PublicConstraints []table.Constraint
```

Comment:
what's the difference between `PublicConstraints` and  `Constraints`

Reply:
maybe should be renamed as `writableConstraints`
the newly added constraint will stay in a state like `write_only`, in this state the constraint will fetch table rows and check it, meantime, the business insert should also follow this constraint but it still not completed.